engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -392,13 +396,15 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
   }
 
   override def getWorkflow(workflowExecutionId: Int)(implicit ec: ExecutionContext): Future[WorkflowDescriptor] = {
-    val action = for {
+    val action: dbio.DBIOAction[WorkflowDescriptor, NoStream, Read with Read] = for {
       workflowExecutionResult <- dataAccess.workflowExecutionsByPrimaryKey(workflowExecutionId).result.head
       workflowAux <- dataAccess.workflowExecutionAuxesByWorkflowExecutionUuid(workflowExecutionResult.workflowExecutionUuid).result.head
-      workflowDescriptor = WorkflowDescriptor(
-        WorkflowId(UUID.fromString(workflowExecutionResult.workflowExecutionUuid)),
-        WorkflowSourceFiles(workflowAux.wdlSource.toRawString, workflowAux.jsonInputs.toRawString, workflowAux.workflowOptions.toRawString)
-      )
+      workflowDescriptor <- materializeWorkflowDescriptor(WorkflowId(UUID.fromString(workflowExecutionResult.workflowExecutionUuid)),
+        WorkflowSourceFiles(workflowAux.wdlSource.toRawString, workflowAux.jsonInputs.toRawString, workflowAux.workflowOptions.toRawString))
+//      workflowDescriptor = WorkflowDescriptor(
+//        WorkflowId(UUID.fromString(workflowExecutionResult.workflowExecutionUuid)),
+//        WorkflowSourceFiles(workflowAux.wdlSource.toRawString, workflowAux.jsonInputs.toRawString, workflowAux.workflowOptions.toRawString)
+//      )
```

Comment:
Yes, I'll revert this file back to it's originality and then modify it, once I know how to act upon it. Was just trying out a couple of things to make for the asynchrony of the WorkflowDescriptor